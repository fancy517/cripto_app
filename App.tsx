import { NavigationContainer } from "@react-navigation/native";
import { PaperProvider } from "react-native-paper";
import StartPage from "./src/views/StartPage";
import TicketDetails from "./src/views/TicketDetails";
import RootStack from "./src/helpers/root.stack";

// const Stack = createNativeStackNavigator();
export default function App() {
  return (
    <NavigationContainer>
      <PaperProvider>
        <RootStack.Navigator initialRouteName="StartPage">
          <RootStack.Screen
            name="StartPage"
            component={StartPage}
            options={{ headerShown: false }}
          />
          <RootStack.Screen
            component={TicketDetails}
            name="TicketDetails"
            options={{ headerShown: false }}
            
          />
        </RootStack.Navigator>
      </PaperProvider>
    </NavigationContainer>
  );
}
