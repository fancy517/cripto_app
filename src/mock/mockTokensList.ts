import Token from "../types/Token";

const TokensList: Token[] = [
  {
    csupply: "",
    id: "1",
    market_cap_usd: "",
    msupply: "",
    name: "2",
    nameid: "",
    percent_change_1h: "",
    percent_change_24h: "",
    percent_change_7d: "",
    price_btc: "",
    price_usd: "",
    rank: 1,
    symbol: "",
    tsupply: "",
    volume24: 1,
    volume24a: 1,
  },
  {
    csupply: "",
    id: "2",
    market_cap_usd: "",
    msupply: "",
    name: "2",
    nameid: "",
    percent_change_1h: "",
    percent_change_24h: "",
    percent_change_7d: "",
    price_btc: "",
    price_usd: "",
    rank: 2,
    symbol: "",
    tsupply: "",
    volume24: 2,
    volume24a: 2,
  },
];

export default TokensList;
