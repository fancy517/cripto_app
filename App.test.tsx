import renderer from "react-test-renderer";
import App from "./App";
import mockTokensList from "./src/mock/mockTokensList";

jest.mock("./src/api", () => {
  return jest.fn().mockImplementation(() => {
    return {
      getAllCriptoTickets: async () => {
        mockTokensList;
      },
    };
  });
});

beforeEach(() => {
  jest.clearAllMocks();
});

describe("<App />", () => {
  it("has 1 child", () => {
    const rawApp = renderer.create(<App />).toJSON();
    expect(rawApp).toMatchSnapshot();
  });
});
