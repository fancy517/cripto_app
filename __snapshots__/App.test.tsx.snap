// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<App /> has 1 child 1`] = `
<RNCSafeAreaProvider
  onInsetsChange={[Function]}
  style={
    {
      "flex": 1,
    }
  }
>
  <View
    collapsable={false}
    pointerEvents="box-none"
    style={
      {
        "flex": 1,
      }
    }
  >
    <View
      style={
        [
          {
            "flex": 1,
          },
          undefined,
        ]
      }
    >
      <RNSScreenStack
        style={
          {
            "flex": 1,
          }
        }
      >
        <RNSScreen
          collapsable={false}
          gestureResponseDistance={
            {
              "bottom": -1,
              "end": -1,
              "start": -1,
              "top": -1,
            }
          }
          nativeBackButtonDismissalEnabled={false}
          onAppear={[Function]}
          onDisappear={[Function]}
          onDismissed={[Function]}
          onHeaderBackButtonClicked={[Function]}
          onNativeDismissCancelled={[Function]}
          onTransitionProgress={[Function]}
          onWillDisappear={[Function]}
          replaceAnimation="push"
          sheetAllowedDetents="large"
          sheetCornerRadius={-1}
          sheetExpandsWhenScrolledToEdge={true}
          sheetGrabberVisible={false}
          sheetLargestUndimmedDetent="all"
          stackPresentation="push"
          style={
            {
              "bottom": 0,
              "left": 0,
              "position": "absolute",
              "right": 0,
              "top": 0,
            }
          }
          swipeDirection="horizontal"
        >
          <RNSScreenStackHeaderConfig
            backButtonInCustomView={false}
            backTitleVisible={true}
            backgroundColor="rgb(255, 255, 255)"
            color="rgb(0, 122, 255)"
            direction="ltr"
            disableBackButtonMenu={false}
            hidden={true}
            hideBackButton={false}
            largeTitleHideShadow={false}
            title="ListOfTokens"
            titleColor="rgb(28, 28, 30)"
            topInsetEnabled={false}
            translucent={false}
          />
          <View
            accessibilityElementsHidden={false}
            importantForAccessibility="auto"
            style={
              {
                "flex": 1,
                "flexDirection": "column-reverse",
              }
            }
          >
            <View
              style={
                [
                  {
                    "flex": 1,
                  },
                  {
                    "backgroundColor": "rgb(242, 242, 242)",
                  },
                  undefined,
                ]
              }
            >
              <View
                style={
                  [
                    {
                      "backgroundColor": "rgba(255, 251, 254, 1)",
                      "elevation": 0,
                      "paddingHorizontal": 0,
                      "paddingTop": 0,
                      "zIndex": 0,
                    },
                    {},
                    {},
                  ]
                }
                testID="appbar-header-root-layer"
              >
                <View
                  collapsable={false}
                  style={
                    {
                      "backgroundColor": "rgba(255, 251, 254, 1)",
                      "height": 64,
                      "shadowColor": "#000",
                      "shadowOffset": {
                        "height": 0,
                        "width": 0,
                      },
                      "shadowOpacity": 0,
                      "shadowRadius": 0,
                    }
                  }
                  testID="appbar-header-outer-layer"
                >
                  <View
                    collapsable={false}
                    style={
                      {
                        "alignItems": "center",
                        "backgroundColor": "rgba(255, 251, 254, 1)",
                        "flex": 1,
                        "flexDirection": "row",
                        "paddingBottom": undefined,
                        "paddingHorizontal": 4,
                        "paddingLeft": undefined,
                        "paddingRight": undefined,
                        "paddingTop": undefined,
                        "shadowColor": "#000",
                        "shadowOffset": {
                          "height": 0,
                          "width": 0,
                        },
                        "shadowOpacity": 0,
                        "shadowRadius": 0,
                      }
                    }
                    testID="appbar-header"
                  >
                    <View
                      collapsable={false}
                      style={
                        {
                          "backgroundColor": "transparent",
                          "borderRadius": 20,
                          "height": 40,
                          "margin": 6,
                          "shadowColor": "#000",
                          "shadowOffset": {
                            "height": 0,
                            "width": 0,
                          },
                          "shadowOpacity": 0,
                          "shadowRadius": 0,
                          "width": 40,
                        }
                      }
                      testID="icon-button-container-outer-layer"
                    >
                      <View
                        collapsable={false}
                        style={
                          {
                            "backgroundColor": "transparent",
                            "borderColor": "rgba(121, 116, 126, 1)",
                            "borderRadius": 20,
                            "borderWidth": 0,
                            "elevation": 0,
                            "flex": 1,
                            "overflow": "hidden",
                            "shadowColor": "#000",
                            "shadowOffset": {
                              "height": 0,
                              "width": 0,
                            },
                            "shadowOpacity": 0,
                            "shadowRadius": 0,
                          }
                        }
                        testID="icon-button-container"
                      >
                        <View
                          accessibilityComponentType="button"
                          accessibilityRole="button"
                          accessibilityState={
                            {
                              "busy": undefined,
                              "checked": undefined,
                              "disabled": true,
                              "expanded": undefined,
                              "selected": undefined,
                            }
                          }
                          accessibilityTraits="button"
                          accessibilityValue={
                            {
                              "max": undefined,
                              "min": undefined,
                              "now": undefined,
                              "text": undefined,
                            }
                          }
                          accessible={true}
                          centered={true}
                          collapsable={false}
                          focusable={true}
                          hitSlop={
                            {
                              "bottom": 6,
                              "left": 6,
                              "right": 6,
                              "top": 6,
                            }
                          }
                          onBlur={[Function]}
                          onClick={[Function]}
                          onFocus={[Function]}
                          onResponderGrant={[Function]}
                          onResponderMove={[Function]}
                          onResponderRelease={[Function]}
                          onResponderTerminate={[Function]}
                          onResponderTerminationRequest={[Function]}
                          onStartShouldSetResponder={[Function]}
                          style={
                            [
                              {
                                "overflow": "hidden",
                              },
                              [
                                {
                                  "alignItems": "center",
                                  "flexGrow": 1,
                                  "justifyContent": "center",
                                },
                                {
                                  "borderRadius": 20,
                                },
                              ],
                            ]
                          }
                          testID="icon-button"
                        >
                          <View
                            style={
                              [
                                {
                                  "alignItems": "center",
                                  "justifyContent": "center",
                                },
                                {
                                  "height": 24,
                                  "width": 24,
                                },
                              ]
                            }
                          >
                            <View
                              collapsable={false}
                              style={
                                {
                                  "bottom": 0,
                                  "left": 0,
                                  "opacity": 1,
                                  "position": "absolute",
                                  "right": 0,
                                  "top": 0,
                                  "transform": [
                                    {
                                      "rotate": "0deg",
                                    },
                                  ],
                                }
                              }
                              testID="cross-fade-icon-current"
                            >
                              <Text />
                            </View>
                          </View>
                        </View>
                      </View>
                    </View>
                    <View
                      pointerEvents="box-none"
                      style={
                        [
                          {
                            "flex": 1,
                            "paddingHorizontal": 12,
                          },
                          {
                            "paddingHorizontal": 0,
                          },
                          [
                            false,
                            {
                              "alignItems": "center",
                            },
                            undefined,
                          ],
                        ]
                      }
                      testID="appbar-content"
                    >
                      <Text
                        accessibilityRole="header"
                        accessibilityTraits="header"
                        accessible={true}
                        numberOfLines={1}
                        style={
                          [
                            {
                              "textAlign": "left",
                            },
                            {
                              "color": "rgba(28, 27, 31, 1)",
                              "writingDirection": "ltr",
                            },
                            [
                              {
                                "fontFamily": "System",
                                "fontSize": 22,
                                "fontWeight": "400",
                                "letterSpacing": 0,
                                "lineHeight": 28,
                              },
                              [
                                {
                                  "color": "rgba(28, 27, 31, 1)",
                                  "fontFamily": "System",
                                  "fontSize": 22,
                                  "fontWeight": "400",
                                  "letterSpacing": 0,
                                  "lineHeight": 28,
                                },
                                false,
                                undefined,
                              ],
                            ],
                          ]
                        }
                        testID="appbar-content-title-text"
                      >
                        Homepage
                      </Text>
                    </View>
                    <View
                      collapsable={false}
                      style={
                        {
                          "backgroundColor": "transparent",
                          "borderRadius": 20,
                          "height": 40,
                          "margin": 6,
                          "shadowColor": "#000",
                          "shadowOffset": {
                            "height": 0,
                            "width": 0,
                          },
                          "shadowOpacity": 0,
                          "shadowRadius": 0,
                          "width": 40,
                        }
                      }
                      testID="icon-button-container-outer-layer"
                    >
                      <View
                        collapsable={false}
                        style={
                          {
                            "backgroundColor": "transparent",
                            "borderColor": "rgba(121, 116, 126, 1)",
                            "borderRadius": 20,
                            "borderWidth": 0,
                            "elevation": 0,
                            "flex": 1,
                            "overflow": "hidden",
                            "shadowColor": "#000",
                            "shadowOffset": {
                              "height": 0,
                              "width": 0,
                            },
                            "shadowOpacity": 0,
                            "shadowRadius": 0,
                          }
                        }
                        testID="icon-button-container"
                      >
                        <View
                          accessibilityComponentType="button"
                          accessibilityRole="button"
                          accessibilityState={
                            {
                              "busy": undefined,
                              "checked": undefined,
                              "disabled": false,
                              "expanded": undefined,
                              "selected": undefined,
                            }
                          }
                          accessibilityTraits="button"
                          accessibilityValue={
                            {
                              "max": undefined,
                              "min": undefined,
                              "now": undefined,
                              "text": undefined,
                            }
                          }
                          accessible={true}
                          centered={true}
                          collapsable={false}
                          focusable={true}
                          hitSlop={
                            {
                              "bottom": 6,
                              "left": 6,
                              "right": 6,
                              "top": 6,
                            }
                          }
                          onBlur={[Function]}
                          onClick={[Function]}
                          onFocus={[Function]}
                          onResponderGrant={[Function]}
                          onResponderMove={[Function]}
                          onResponderRelease={[Function]}
                          onResponderTerminate={[Function]}
                          onResponderTerminationRequest={[Function]}
                          onStartShouldSetResponder={[Function]}
                          style={
                            [
                              {
                                "overflow": "hidden",
                              },
                              [
                                {
                                  "alignItems": "center",
                                  "flexGrow": 1,
                                  "justifyContent": "center",
                                },
                                {
                                  "borderRadius": 20,
                                },
                              ],
                            ]
                          }
                          testID="icon-button"
                        >
                          <View
                            style={
                              [
                                {
                                  "alignItems": "center",
                                  "justifyContent": "center",
                                },
                                {
                                  "height": 24,
                                  "width": 24,
                                },
                              ]
                            }
                          >
                            <View
                              collapsable={false}
                              style={
                                {
                                  "bottom": 0,
                                  "left": 0,
                                  "opacity": 1,
                                  "position": "absolute",
                                  "right": 0,
                                  "top": 0,
                                  "transform": [
                                    {
                                      "rotate": "0deg",
                                    },
                                  ],
                                }
                              }
                              testID="cross-fade-icon-current"
                            >
                              <Text />
                            </View>
                          </View>
                        </View>
                      </View>
                    </View>
                  </View>
                </View>
              </View>
              <View
                style={
                  {
                    "flex": 1,
                  }
                }
              >
                <RCTScrollView
                  data={[]}
                  getItem={[Function]}
                  getItemCount={[Function]}
                  keyExtractor={[Function]}
                  onContentSizeChange={[Function]}
                  onEndReached={[Function]}
                  onEndReachedThreshold={0.1}
                  onLayout={[Function]}
                  onMomentumScrollBegin={[Function]}
                  onMomentumScrollEnd={[Function]}
                  onScroll={[Function]}
                  onScrollBeginDrag={[Function]}
                  onScrollEndDrag={[Function]}
                  removeClippedSubviews={false}
                  renderItem={[Function]}
                  scrollEventThrottle={50}
                  stickyHeaderIndices={[]}
                  style={
                    {
                      "flex": 1,
                    }
                  }
                  viewabilityConfigCallbackPairs={[]}
                >
                  <View />
                </RCTScrollView>
              </View>
            </View>
          </View>
        </RNSScreen>
      </RNSScreenStack>
    </View>
  </View>
</RNCSafeAreaProvider>
`;
